js/src/main/scala/crystal/react/hooks/UseDebouncedTimeout.scala

```diff
@@ -0,0 +1,67 @@
+package crystal.react.hooks
+
+import cats.effect.Fiber
+import cats.syntax.all._
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.hooks.CustomHook
+import japgolly.scalajs.react.util.DefaultEffects.{ Async => DefaultA }
+
+import scala.concurrent.duration.FiniteDuration
+
+/** Given a `timeout`, this hooks provide a `TimeoutHandle` which can be used to submit an effect,
+  * which will be run after the `timeout`. If a new effect is submitted before the `timeout`, the
+  * previous one will be canceled (whether it already started running or not).
+  *
+  * A submitted effect can be explicitly canceled.
+  */
+object UseDebouncedTimeout {
+
+  // Changing duration while component is mounted is not supported.
+  val hook = CustomHook[FiniteDuration]
+    .useRef(none[Fiber[DefaultA, Throwable, Unit]])
+    .useMemoBy((_, _) => ())((duration, timerRef) => _ => new TimeoutHandle(duration, timerRef))
+    .buildReturning((_, _, timeoutHandle) => timeoutHandle)
+
+  object HooksApiExt {
```

Comment:
I guess this is you custom hook?

Reply:
Right, this is to have nice syntax `ScalaFnComponent.withHooks[Props].useOurCustomHook`